ocrd_tesserocr/segment_table.py

```diff
@@ -0,0 +1,245 @@
+from __future__ import absolute_import
+
+import os.path
+from tesserocr import (
+    PyTessBaseAPI,
+    PSM, RIL, PT
+)
+
+from ocrd_utils import (
+    getLogger,
+    concat_padded,
+    coordinates_for_segment,
+    polygon_from_x0y0x1y1,
+    points_from_polygon,
+    MIMETYPE_PAGE,
+    membername
+)
+from ocrd_modelfactory import page_from_file
+from ocrd_models.ocrd_page import (
+    MetadataItemType,
+    LabelsType, LabelType,
+    CoordsType,
+    TextRegionType,
+    to_xml)
+from ocrd_models.ocrd_page_generateds import (
+    TableRegionType,
+    TextTypeSimpleType,
+    RegionRefType,
+    RegionRefIndexedType,
+    OrderedGroupType,
+    OrderedGroupIndexedType,
+    UnorderedGroupType,
+    UnorderedGroupIndexedType,
+    ReadingOrderType
+)
+from ocrd import Processor
+
+from .config import TESSDATA_PREFIX, OCRD_TOOL
+
+TOOL = 'ocrd-tesserocr-segment-table'
+LOG = getLogger('processor.TesserocrSegmentTable')
+
+class TesserocrSegmentTable(Processor):
+
+    def __init__(self, *args, **kwargs):
+        kwargs['ocrd_tool'] = OCRD_TOOL['tools'][TOOL]
+        kwargs['version'] = OCRD_TOOL['version']
+        super(TesserocrSegmentTable, self).__init__(*args, **kwargs)
+
+    def process(self):
+        """Performs table cell segmentation with Tesseract on the workspace.
+        
+        Open and deserialize PAGE input files and their respective images,
+        then iterate over the element hierarchy down to the block level
+        for table regions. If ``overwrite_regions`` is enabled and any
+        layout annotation already exists inside, then remove it.
+        
+        Set up Tesseract to detect text blocks (as table cells).
+        (This is not Tesseract's internal table structure recognition,
+        but the general page segmentation.)
+        Add each to the block at the detected coordinates.
+        
+        Produce a new output file by serialising the resulting hierarchy.
+        """
+        overwrite_regions = self.parameter['overwrite_regions']
+        
+        with PyTessBaseAPI(path=TESSDATA_PREFIX) as tessapi:
+            # disable table detection here, so tables will be
+            # analysed as independent text/line blocks:
+            tessapi.SetVariable("textord_tabfind_find_tables", "0")
+            # this should yield additional blocks within the table blocks
+            # from the page iterator, but does not in fact (yet?):
+            #tessapi.SetVariable("textord_tablefind_recognize_tables", "1")
```

Comment:
Since this is copied here, maybe extend the comment here as well.

Reply:
cf 
66da5f6